src/util/validate_types.h

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************\
+
+Module: Validate types
+
+Author: Daniel Poetzl
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_VALIDATE_TYPES_H
+#define CPROVER_UTIL_VALIDATE_TYPES_H
+
+class typet;
+class namespacet;
+enum class validation_modet;
+
+void check_type_pick(const typet &type, const validation_modet vm);
```

Comment:
Parameter names could be omitted in the declarations. It's obvious what they are.